sdk/core/azure-core/src/main/java/com/azure/core/util/SharedExecutorService.java

```diff
@@ -0,0 +1,350 @@
+    private static final ReflectiveInvoker SET_VIRTUAL_THREAD_BUILDER_THREAD_NAME;
+    private static final ReflectiveInvoker CREATE_VIRTUAL_THREAD_FACTORY;
+
+    static {
+        boolean virtualThreadSupported;
+        ReflectiveInvoker getVirtualThreadBuilder;
+        ReflectiveInvoker setVirtualThreadBuilderThreadName;
+        ReflectiveInvoker createVirtualThreadFactory;
+
+        try {
+            getVirtualThreadBuilder = ReflectionUtils.getMethodInvoker(null,
+                Class.forName("java.lang.Thread").getDeclaredMethod("ofVirtual"));
+            setVirtualThreadBuilderThreadName = ReflectionUtils.getMethodInvoker(null,
+                Class.forName("java.lang.Thread$Builder").getDeclaredMethod("name", String.class));
+            createVirtualThreadFactory = ReflectionUtils.getMethodInvoker(null,
+                Class.forName("java.lang.Thread$Builder").getDeclaredMethod("factory"));
+            virtualThreadSupported = true;
+        } catch (Exception | LinkageError e) {
+            virtualThreadSupported = false;
+            getVirtualThreadBuilder = null;
+            setVirtualThreadBuilderThreadName = null;
+            createVirtualThreadFactory = null;
+        }
+
+        VIRTUAL_THREAD_SUPPORTED = virtualThreadSupported;
+        GET_VIRTUAL_THREAD_BUILDER = getVirtualThreadBuilder;
+        SET_VIRTUAL_THREAD_BUILDER_THREAD_NAME = setVirtualThreadBuilderThreadName;
+        CREATE_VIRTUAL_THREAD_FACTORY = createVirtualThreadFactory;
+    }
+
+    private final AtomicReference<ExecutorWithMetadata> wrappedExecutorService;
+
+    private SharedExecutorService() {
+        this.wrappedExecutorService = new AtomicReference<>();
+    }
+
+    /**
+     * Gets the shared instance of the executor service.
+     *
+     * @return The shared instance of the executor service.
+     */
+    public static SharedExecutorService getInstance() {
+        return INSTANCE;
+    }
+
+    /**
+     * Sets the backing executor service for the shared instance.
+     * <p>
+     * This updates the executor service for all users of the {@link #getInstance() shared instance}. Meaning, if
+     * another area in code already had a reference to the shared instance, it will now use the passed executor service
+     * to execute tasks.
+     * <p>
+     * If the executor service is already set, this will replace it with the new executor service. If the replaced
+     * executor service was created by this class, it will be shut down.
+     * <p>
+     * If the passed executor service is null, this will throw a {@link NullPointerException}. If the passed executor
+     *
+     * @param executorService The executor service to set as the shared instance.
+     * @throws NullPointerException If the passed executor service is null.
+     * @throws IllegalStateException If the passed executor service is shutdown or terminated.
+     */
+    public static void setExecutorService(ExecutorService executorService) {
```

Comment:
what would be the use case to set an executor? Is it necessary to have it publicly settable?

Reply:
Scenario where a customer wants to inject their own executor service to manage running asynchronous code in the SDKs. Similar to what Reactor allows with their Schedulers where a custom implementation can be injected to be used as the global value.

I'll add more comments.